common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -37,61 +37,51 @@ public final class ConfigManager extends Manager {
     private static final String OVERLAY_GROUPS_JSON_KEY = "overlayGroups";
     private static final Set<ConfigHolder> CONFIG_HOLDERS = new TreeSet<>();
 
-    private static final List<ConfigHolder> OVERLAY_GROUP_CONFIG_HOLDERS = new ArrayList<>();
-    private static final List<OverlayGroupHolder> OVERLAY_GROUP_FIELDS = new ArrayList<>();
-
     private final File userConfig;
     private JsonObject configObject;
 
-    public ConfigManager(ConfigUpfixerManager configUpfixerManager, JsonManager jsonManager) {
-        super(List.of(configUpfixerManager, jsonManager));
+    public ConfigManager(
+            ConfigUpfixerManager configUpfixerManager,
+            JsonManager jsonManager,
+            FeatureManager feature,
+            OverlayManager overlay) {
+        super(List.of(configUpfixerManager, jsonManager, feature, overlay));
 
         userConfig = new File(CONFIGS, McUtils.mc().getUser().getUuid() + FILE_SUFFIX);
+    }
 
+    public void init() {
         // First, we load the config file
         configObject = Managers.Json.loadPreciousJson(userConfig);
 
         // Now, we have to apply upfixers, before any config loading happens
-        if (configUpfixerManager.runUpfixers(configObject)) {
+        if (Managers.ConfigUpfixer.runUpfixers(configObject)) {
             Managers.Json.savePreciousJson(userConfig, configObject);
         }
-    }
-
-    public void registerFeature(Feature feature) {
-        registerOverlayGroups(feature);
-
-        for (Overlay overlay : feature.getOverlays()) {
-            registerConfigOptions(overlay);
-        }
 
-        registerConfigOptions(feature);
-    }
+        // Register all features and overlays
+        Managers.Feature.getFeatures().forEach(this::registerFeature);
 
-    private void registerOverlayGroups(Feature feature) {
-        List<OverlayGroupHolder> holders = Stream.of(feature.getClass().getDeclaredFields())
-                .filter(f -> f.isAnnotationPresent(OverlayGroup.class))
-                .map(field -> new OverlayGroupHolder(
-                        field, feature, field.getAnnotation(OverlayGroup.class).instances()))
-                .toList();
+        // Finish off the config init process
 
-        if (holders.isEmpty()) return;
+        // Load configs for all features
+        Managers.Config.reloadConfiguration();
 
-        OVERLAY_GROUP_FIELDS.addAll(holders);
-        feature.addOverlayGroups(holders);
+        // Save config file after loading all configurables' options
```

Comment:
Is it still needed to both load and then save configs directly thereafter? How come?

Reply:
Nothing directly changed about overlay group config loading, it was just a cleanup. The config cleanup will be separate, and I have yet to fully figure it out. But basically this code was moved here from featuremanager, as we now load configs after feature manager.